systems/plants/@RigidBodyManipulator/manipulatorDynamics.m

```diff
@@ -31,6 +25,11 @@
   [f_ext, B] = computeExternalForcesAndInputMatrix(obj, q, v);
 end
 
+options.use_mex = use_mex;
+options.compute_gradients = compute_gradients;
+options.compute_JdotV = true;
```

Comment:
You don't need it for H (H is dependent on q only after all), but you need Jdot * v for C.
I interpret C as the output of an inverse dynamics algorithm with vdot set to zero. The inverse dynamics algorithm that we use is the recursive newton-euler algorithm. It basically has three steps:

1. Starting from the root of the tree moving towards the leaves, compute the spatial accelerations of all bodies w.r.t. the world
1. Starting from the leaves and moving towards the root, compute the net wrench exerted on each body needed to achieve the spatial acceleration computed in the previous step. Subtract off the wrenches already exerted on the body by children joints to arrive at the wrench that should be exerted across the body's parent joint.
1. For each body, premultiply the joint wrench by the transpose of the motion subspace to arrive at the joint torque that needs to be exerted.

The Jdot * v entries in our kinematics cache are exactly the spatial accelerations of all of the bodies w.r.t. the world, expressed in world frame, when the joint accelerations are set to zero, i.e. the output of step 1 above.